include/swupd.h

```diff
@@ -104,6 +105,7 @@ struct file {
 
 	char *staging; /* output name used during download & staging */
 	CURL *curl;    /* curl handle if downloading */
+	int fd;        /* file written into during downloading, unset when fd_valid is false */
```

Comment:
I would prefer to use fd == -1 to indicate an invalid file descriptor, with the requisite addition of struct file initializers.